src/services/refactors/convertImport.ts

```diff
@@ -138,13 +138,36 @@ namespace ts.refactor {
+            const symbol = checker.getSymbolAtLocation(namedImport.name);
+            if (symbol) {
+                toConvertSymbols.add(symbol);
+            }
+        });
         const preferredName = moduleSpecifier && isStringLiteral(moduleSpecifier) ? codefix.moduleSpecifierToValidIdentifier(moduleSpecifier.text, ScriptTarget.ESNext) : "module";
-        const namespaceNameConflicts = toConvert.elements.some(element =>
-            FindAllReferences.Core.eachSymbolReferenceInFile(element.name, checker, sourceFile, id =>
-                !!checker.resolveName(preferredName, id, SymbolFlags.All, /*excludeGlobals*/ true)) || false);
+        function hasNamespaceNameConflict(namedImport: ImportSpecifier): boolean {
+            // We need to check if the preferred namespace name (`preferredName`) we'd like to use in the refactored code will present a name conflict.
+            // A name conflict means that, in a scope where we would like to use the preferred namespace name, there already exists a symbol with that name in that scope.
+            // We are going to use the namespace name in the scopes the named imports being refactored are referenced,
+            // so we look for conflicts by looking at every reference to those named imports.
+            return !!FindAllReferences.Core.eachSymbolReferenceInFile(namedImport.name, checker, sourceFile, id => {
+                const symbol = checker.resolveName(preferredName, id, SymbolFlags.All, /*excludeGlobals*/ true);
+                if (symbol) { // There already is a symbol with the same name as the preferred namespace name.
+                    if (toConvertSymbols.has(symbol)) { // `preferredName` resolves to a symbol for one of the named import references we are going to transform into namespace import references...
+                        return isExportSpecifier(id.parent); // ...but if this reference is an export specifier, it will not be transformed, so it is a conflict; otherwise, it will be renamed and is not a conflict.
+                    }
+                    return true; // `preferredName` resolves to any other symbol, which will be present in the refactored code and so poses a name conflict.
+                }
+                return false; // There is no symbol with the same name as the preferred namespace name, so no conflict.
+            });
+        }
+        const namespaceNameConflicts = toConvert.elements.some(hasNamespaceNameConflict);
         const namespaceImportName = namespaceNameConflicts ? getUniqueName(preferredName, sourceFile) : preferredName;
 
-        const neededNamedImports: ImportSpecifier[] = [];
+        // Imports that need to be kept as named imports in the refactored code, to avoid changing the semantics.
+        // More specifically, those are named imports that appear in named exports in the original code, e.g. `a` in `import { a } from "m"; export { a }`.
+        const neededNamedImports: Set<ImportSpecifier> = new Set();

```

Comment:
Changed to use a set instead of an array. The old code used the array as a set by comparing the named imports' `name` properties (of type `Identifiers`) (`element.name` in for loop below), but it pushed into the array a node with a newly-created `Identifier` as `name`, so in fact this new node's `name` would never be equal to the named imports' `name`, and therefore the same named import could be repeated, as in:
```ts
import { a, b } from "m";
export { b };
export { b as c };
```
would become
```ts
import * as m from "m";
import { b, b } from "m"; // Duplicate b import
export { b };
export { b as c };
```